azkaban-web-server/src/main/java/cloudflow/services/ExecutionServiceImpl.java

```diff
@@ -101,6 +104,70 @@ private ExecutionBasicResponse executionResponseFromFlow(ExecutableFlow executab
         .collect(Collectors.toList());
   }
 
+  // todo(sshardool): this should have a check for run-away recursion on malformed input
+  private static ExecutionNodeResponse nodeResponseFromExecutionNode(
+      ExecutableNode executableNode) {
+    ExecutionNodeResponseBuilder nodeBuilder = ExecutionNodeResponseBuilder.newBuilder();
+    nodeBuilder = nodeBuilder.withNodeId(executableNode.getId())
+        .withStartTime(executableNode.getStartTime())
+        .withEndTime(executableNode.getEndTime())
+        .withUpdateTime(executableNode.getUpdateTime())
+        .withNodeType(executableNode.getType())
+        .withCondition(executableNode.getCondition())
+        .withNestedId(executableNode.getNestedId());
+
+    if (executableNode.getInNodes() != null) {
+      nodeBuilder = nodeBuilder.withInputNodeIds(new ArrayList<>(executableNode.getInNodes()));
+    }
+
+    if (executableNode instanceof ExecutableFlowBase) {
+      ExecutableFlowBase flowBase = (ExecutableFlowBase) executableNode;
+      List<ExecutionNodeResponse> subNodeList = new ArrayList<>();
+
+      for (ExecutableNode subNode : flowBase.getExecutableNodes()) {
+        ExecutionNodeResponse subNodeResponse = nodeResponseFromExecutionNode(subNode);
+        subNodeList.add(subNodeResponse);
+      }
+
+      nodeBuilder = nodeBuilder.withNodeList(subNodeList)
+          .withBaseFlowId(flowBase.getFlowId());
+    }
+
+    return nodeBuilder.build();
+  }
+
+
+  @Override
+  public ExecutionDetailedResponse getSingleExecution(String executionIdString) {
+    requireNonNull(executionIdString, "execution id is null");
+    int executionId;
+    String errorMessage;
+    try {
+      executionId = Integer.parseInt(executionIdString);
+    } catch (NumberFormatException nfe) {
+      errorMessage = "Execution Id must be an integer";
+      logger.error(errorMessage, nfe);
+      throw new CloudFlowException(errorMessage, nfe);
```

Comment:
This looks like a validation exception. Can you handle those and CloudFlowException(500 errors) in the servlet? Currently only not found exceptions are handled.

Reply:
Makes sense. However I would prefer to make that change separately, such that all the `get execution` apis are covered by it rather than having a try-catch block for each flavor.

Reply:
Why do you prefer to throw a NumberFormatException instead of a CloudFlowValidationException?